File: Typescript/Q2341-Maximum-Number-of-Pairs-in-Array.ts
function numberOfPairs(nums: number[]): number[] {
    if(nums.length==0){
               let ar=[0,0];
           }
           let pair=0;
           let left =0;
           let table=new Map();
           for(let i=0;i<nums.length;i++){
               if(table.get(nums[i])!=null){
                   let occur=table.get(nums[i]);
                   occur++;
                   pair+=parseInt(`${occur/2}`);
                   occur%=2;
                   table.set(nums[i],occur);
               }else{
                   table.set(nums[i],1);
               }
           }
            table.forEach((value,key)=>{
                left+=value;
            });
           let ar=[pair,left];
           return ar;
   };

File: Java/Q2341-Maximum-Number-of-Pairs-in-Array.java
class Solution {
    public int[] numberOfPairs(int[] nums) {
        if(nums.length==0){
            int[] ar={0,0};
        }
        int pair=0;
        int left =0;
        HashMap<Integer,Integer> table=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(table.get(nums[i])!=null){
                int occur=table.get(nums[i]);
                occur++;
                pair+=occur/2;
                occur%=2;
                table.put(nums[i],occur);
            }else{
                table.put(nums[i],1);
            }
        }
         for (Map.Entry<Integer,Integer> entry : table.entrySet()) {
           left+=entry.getValue();
         }
        int[] ar={pair,left};
        return ar;
    }
}